connect_deprish_to_ferc1.py

```diff
@@ -682,14 +740,79 @@ def split_ferc1_data_on_split_cols(self,
 
         # merge in the newly generated split/assigned data column
         df_final = pd.merge(
-            same_smol, df_w_tots[
-                idx_cols_ferc1 + ['record_id_eia_deprish', new_data_col]])
+            same_smol,
+            df_w_tots[idx_cols_ferc1 + ['record_id_eia_deprish',
+                                        new_data_col]].drop_duplicates(),
+        )
         return df_final
 
+    def agg_ferc_data_cols(self):
+        """
+        Aggregate smaller level FERC1 data columns to depreciation records.
+
+        When the depreciation matches are at a higher level than the FERC1,
+        this method aggregates the many FERC1 records into the level of the
+        depreciation records.
+        """
+        data_cols_own_frac = [
+            f"{col}_own_frac" for col in DATA_COLS_TO_SPLIT.keys()]
+        same_beeg = self.matches_df.loc[
+            (self.matches_df.match_method == 'same_quals')
+            & (self.matches_df.level_deprish == 'beeg')
+        ]
+
+        # dict to rename the summed data columns to their new name
+        rename_dict = {}
+        for data_col_of, data_col in zip(data_cols_own_frac,
+                                         DATA_COLS_TO_SPLIT):
+            rename_dict[data_col_of] = self._get_clean_new_data_col(data_col)
+        # sum the data columns at the level of the depreciation records
+        same_beeg_sum = (
+            same_beeg.groupby('record_id_eia_deprish')
+            [data_cols_own_frac]
+            .sum()
+            .reset_index()
+            .rename(columns=rename_dict)
+        )
+
+        # squish the new summed columns back into the og df
+        same_beeg = pd.merge(same_beeg,
+                             same_beeg_sum,
+                             on=['record_id_eia_deprish'],
+                             how='left')
+        return same_beeg
+
     def _get_clean_new_data_col(self, data_col):
         # some of the data columns already have a ferc1 suffix because the same
         # column name also shows up in the EIA data... so we want to remove the
         # double ferc1 if it shows up
         new_data_col = f"{data_col}_ferc1_deprish"
         new_data_col = new_data_col.replace("ferc1_ferc1", "ferc1")
         return new_data_col
+
+    def test_same_true_fraction_owned(self, same_true):
+        """
+        Test the same_true scaling.
+
+        Raises:
+            AssertionError: If there is some error in the standard scaling with
+            fraction_owned columns.
+        """
+        for data_col in DATA_COLS_TO_SPLIT:
+            new_data_col = self._get_clean_new_data_col(data_col)
+            not_same = same_true[
+                (same_true.match_method == 'same_true')
+                & ~(same_true[data_col] == same_true[new_data_col])
+                & (same_true[data_col].notnull())
+                & (same_true[new_data_col].notnull())
+            ]
+            if len(not_same) != 0:
```

Comment:
Empty sequences evaluate to `False` and PEP8 recommends using the sequence itself rather than its length. So this could be just `if not_same:`

Reply:
ah great!